torch/nn/functional.py

```diff
@@ -3672,35 +3672,18 @@ def upsample(input, size=None, scale_factor=None, mode="nearest", align_corners=
 
 upsample.__doc__ = upsample.__doc__.format(**reproducibility_notes)
 
+_InterpolateSizeType = Union[None, int, Tuple[int], Tuple[int, int], Tuple[int, int, int]]
+_InterpolateScaleType = Union[None, float, Tuple[float], Tuple[float, float], Tuple[float, float, float]]
```

Comment:
Is this actually limited to 3D? Or should it be `Tuple[float, ...]` like the doc?

Reply:
According to a docstring, it only supports 1–3 spatial dimensions, so `Tuple[float, ...]` is strictly-speaking incorrect but might be simpler. What do you think?

Reply:
Ok, it is actually checking for that. So let's go with the specific spec as you did here and update the docstring.
I wish there was a less verbose way to do this though...